x-pack/plugins/enterprise_search/public/applications/app_search/components/meta_engine_creation/meta_engine_creation_logic.ts

```diff
@@ -0,0 +1,131 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { generatePath } from 'react-router-dom';
+
+import { kea, MakeLogicType } from 'kea';
+
+import { Meta } from '../../../../../common/types';
+import { DEFAULT_META } from '../../../shared/constants';
+import { flashAPIErrors, setQueuedSuccessMessage } from '../../../shared/flash_messages';
+import { HttpLogic } from '../../../shared/http';
+import { KibanaLogic } from '../../../shared/kibana';
+import { ENGINE_PATH } from '../../routes';
+import { formatApiName } from '../../utils/format_api_name';
+import { EngineDetails } from '../engine/types';
+
+import { META_ENGINE_CREATION_SUCCESS_MESSAGE } from './constants';
+
+interface MetaEngineCreationValues {
+  indexedEngineNames: string[];
+  name: string;
+  rawName: string;
+  selectedIndexedEngineNames: string[];
+}
+
+interface MetaEngineCreationActions {
+  fetchIndexedEngineNames(page?: number): { page: number };
+  onEngineCreationSuccess(): void;
+  setIndexedEngineNames(
+    indexedEngineNames: MetaEngineCreationValues['indexedEngineNames']
+  ): { indexedEngineNames: MetaEngineCreationValues['indexedEngineNames'] };
+  setRawName(rawName: string): { rawName: string };
+  setSelectedIndexedEngineNames(
+    selectedIndexedEngineNames: MetaEngineCreationValues['selectedIndexedEngineNames']
+  ): { selectedIndexedEngineNames: MetaEngineCreationValues['selectedIndexedEngineNames'] };
+  submitEngine(): void;
+}
+
+export const MetaEngineCreationLogic = kea<
+  MakeLogicType<MetaEngineCreationValues, MetaEngineCreationActions>
+>({
+  path: ['enterprise_search', 'app_search', 'meta_engine_creation_logic'],
+  actions: {
+    fetchIndexedEngineNames: (page = DEFAULT_META.page.current) => ({ page }),
+    onEngineCreationSuccess: true,
+    setIndexedEngineNames: (indexedEngineNames) => ({ indexedEngineNames }),
+    setRawName: (rawName) => ({ rawName }),
+    setSelectedIndexedEngineNames: (selectedIndexedEngineNames) => ({ selectedIndexedEngineNames }),
+    submitEngine: () => null,
+  },
+  reducers: {
+    indexedEngineNames: [
+      [],
+      {
+        setIndexedEngineNames: (_, { indexedEngineNames }) => indexedEngineNames,
+      },
+    ],
+    rawName: [
```

Comment:
[not a blocker/change request] Just curious, why put this in-between `indexedEngineNames` and `selectedIndexedEngineNames`? Those feel like they conceptually belong together/next to each other 🤷

Reply:
I always prefer objective alphabetization over subjective conceptual groupings